tests/TestCase/Identifier/LdapIdentifierTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Test\TestCase\Identifier;
+
+use Authentication\Identifier\LdapIdentifier as BaseLdapIdentifier;
+use Authentication\Test\TestCase\AuthenticationTestCase as TestCase;
+use Cake\Datasource\EntityInterface;
+
+/**
+ * Overwrite all LdapOopTrait methods to enable tests
+ */
+class LdapIdentifier extends BaseLdapIdentifier
```

Comment:
Same as above. It makes little sense implementing an OOP wrapper as a trait and then using this trait in a mock class for tests.
This wrapper could be implemented as a regular class and injected as a dependency to the identifier or build from config array.